### PR TITLE
Why don't we?

### PR DESCRIPTION
**Pull request by KillWolfVlad**
GitKraken on system electron v1.8.2 not work (infinitely refreshes), so I used electron from github

### CODE REVIEW

File: PKGBUILD [hunk 1]
```diff
@@ -5,43 +5,74 @@
...
-pkgdesc="The intuitive, fast, and beautiful cross-platform Git client."
-url="https://www.gitkraken.com/"
+pkgdesc='The legendary Git GUI client for Arch Linux'
+url='https://www.gitkraken.com/'
 provides=('gitkraken')
 arch=('x86_64')
 license=('custom')
-depends=('gtk2' 'nss' 'libxtst' 'libgnome-keyring' 'gconf' 'alsa-lib' 'libcurl-gnutls' 'libxss' 'rtmpdump')
+depends=('gtk2' 'nss' 'libxtst' 'libgnome-keyring' 'gconf' 'alsa-lib' 'libxss' 'rtmpdump')
+makedepends=('nodejs' 'npm' 'python2' 'yarn' 'asar' 'jq' 'libxkbfile')
```
Comment (Azd325) [inline PKGBUILD]: Why you need to install npm?

File: PKGBUILD [hunk 2]
```diff
@@ -5,43 +5,74 @@
...
 
+_electron=1.6.17
+
 pkgname=gitkraken
-pkgrel=2
+pkgrel=3
 pkgver=3.3.4
-pkgdesc="The intuitive, fast, and beautiful cross-platform Git client."
-url="https://www.gitkraken.com/"
+pkgdesc='The legendary Git GUI client for Arch Linux'
```
Comment (Azd325) [inline PKGBUILD]: unnecessary

File: PKGBUILD (hunk 1 above)
Comment (Azd325) [reply PKGBUILD]: Why you need to install python2?

Comment (Azd325) [reply PKGBUILD]: It would be also could to explain every `makedepends` somewhere.

File: PKGBUILD [hunk 3]
```diff
@@ -5,43 +5,74 @@
...
+  chmod 755 "${pkgdir}/opt/gitkraken/electron"
 
-    install -D -m755 "./gitkraken.sh" "${pkgdir}/usr/bin/gitkraken"
-    install -D -m644 "./eula.html" "${pkgdir}/usr/share/licenses/${pkgname}/eula.html"
-    install -D -m644 "./GitKraken.desktop" "${pkgdir}/usr/share/applications/GitKraken.desktop"
-    install -D -m644 "./gitkraken.png" "${pkgdir}/usr/share/pixmaps/gitkraken.png"
+  install -D -m644 "./eula.html" "${pkgdir}/usr/share/licenses/${pkgname}/eula.html"
+  install -D -m644 "./gitkraken.desktop" "${pkgdir}/usr/share/applications/gitkraken.desktop"
+  install -D -m644 "./gitkraken.png" "${pkgdir}/usr/share/pixmaps/gitkraken.png"
+  install -D -m755 "./gitkraken.sh" "${pkgdir}/usr/bin/gitkraken"
```
Comment (Azd325) [inline PKGBUILD]: unnecessary changes.

Comment (KillWolfVlad) [reply PKGBUILD]: fixed

File: PKGBUILD (hunk 2 above)
Comment (KillWolfVlad) [reply PKGBUILD]: fixed

**Comment by KillWolfVlad** [comment]
> What I don't like that is that it seems like a dev build

No, it is release build.

Total Installed Size | Rebuild
------------------------- | ----------
235,17 MiB             | No
236,38 MiB             | Yes

Difference: +1,21 MiB

> Also this build size and time are huge for something what runs currently complete from the arch packages repo.

I agree, rebuild is not easy job for PC.

> Is there also a different size of the executable size and is there a difference in consumption of memory and cpu?

I think the differences are minor.

> Do you know if there is a way of providing two install methods in one PKGBUILD like one normal and one to compile?

Fixed.

> It would be also could to explain every `makedepends` somewhere.

- `npm` and `python2` - node-gyp
- `yarn` - Fast, reliable, and secure dependency management
- `asar` - unpack and pack `app.asar`
- `jq` - parse `package.json`
- `libxkbfile` - required by [keyboard-layout](https://github.com/atom/keyboard-layout)

**Comment by MarcinWieczorek** [comment]
>Do you know if there is a way of providing two install methods in one PKGBUILD like one normal and one to compile?
There is no pretty solution.

I asked `Why don't we?`, because I'd like to see Gitkraken running on system electron. The rebuild process might be costly, but that might give the package a chance to get moved to community. I got one electron based package moved to the repos already. I'm worried by the license tho...
Can we make Gitkraken weigh less? How many dependencies can we provide externally?

**Comment by KillWolfVlad** [comment]
> Gitkraken running on system electron

GitKraken does not support electron v1.8.x (bundled with v1.6.11)

You can test this [branch](https://github.com/KillWolfVlad/GitKraken-Pro-AUR/tree/why-dont-we-system-electron)

> moved to community

It would be great!

If @Axosoft will upgrade GitKraken to support electron v1.8.x (v2.x coming soon) we able to can use

system electron + app.asar (from GitKraken) as well (perhaps even without rebuild)

> I'm worried by the license tho...

I think only `app.asar` is proprietary, `electron` as runtime is open source

> Can we make Gitkraken weigh less?

New install size = Old install size - bundled electron

> How many dependencies can we provide externally?

It is difficult to determine. Let them remain unchanged.

**Comment by KillWolfVlad** [comment]
GitKraken v3.6.0 bundled with electron v1.8.4 don't work (infinity reload) with system electron v1.8.4, but successfully work with electron v1.8.6 from GitHub.
